test/integration/base_integration_test.cc

```diff
@@ -47,6 +48,7 @@ BaseIntegrationTest::BaseIntegrationTest(const InstanceConstSharedPtrFn& upstrea
       version_(version), upstream_address_fn_(upstream_address_fn),
       config_helper_(version, *api_, config),
       default_log_level_(TestEnvironment::getOptions().logLevel()) {
+  Envoy::Server::validateProtoDescriptors();
```

Comment:
Would like to highlight 2 questions regarding this:
1. Is calling this outside of the mutex ok?
2. Could there be a use of one of the validated protos before the BaseIntergrationTest c'tor is executed?

I'm less worried about 2 because this change only impacts tests, and so I guess we will see failures.
For 1, I'm guessing this is only relevant to multi-threaded use-cases, and AFAIK threading only starts after the integration test is instantiated.

Feel free to correct me if I'm wrong on any of the above.

Reply:
1) I think we're largely checking things that are registered at or before start up, so this is fine
2) yes, but then the test will fail with a slightly less helpful warning so if the goal is to fail (it's an ASSERT not RELEASE_ASSERT) we will achieve the goal just less helpfully :-)